### DISCUSSION TITLE
Python version support

### DISCUSSION THREAD

**Discussion by artofscience** [artofscience/SAOR#68] [General]
Hi there, I added a Workflow that runs the tests via `python -m pytest tests/` for Python versions 3.8 and 3.9. I found we included some features that are not supported by versions < 3.8. What version do we intend to support?

**Comment by MaxvdKolk** [comment]
From 3.8 and upwards is probably fine. Most systems come with >=3.8 now I think, at least Ubuntu and MacOS. Supporting lower versions is probably not too hard, we might just need to change the f-string prints `f'{variable}'` and some use of the walrus operator `:=` at some places.
